vm/src/obj/objmodule.rs

```diff
@@ -34,6 +36,17 @@ impl PyModuleRef {
         }
     }
 
+    fn dict(self, vm: &VirtualMachine) -> PyResult<PyDictRef> {
+        let name_obj = vm.new_str(self.name.clone());
+        if let Some(ref dict) = &self.into_object().dict {
+            let mod_dict = dict.clone();
+            mod_dict.set_item("__name__", name_obj, vm)?;
```

Comment:
This set_item will happen each time the dict is retrieved. Is there a way to create the dict when the module is constructed? So in init or new?

Reply:
Fixed